### PR TITLE
Closes #2664 bug fix for dataset_ref in some functions

### LINKED ISSUE #2664: Bug: derive_summar_records() add variables from dataset_ref
### What happened?

If a variable is in both `dataset_add` and `dataset_ref`, it is added to the new records even if it is not in `by_vars`. The variable is populated only for records which originates from `dataset_ref` (see example below).

`derive_summary_records()` should be updated such that only `by_vars` are kept from `dataset_ref`.

### Session Information

_No response_

### Reproducible Example

```
library(tibble)
ref_data <- tibble(USUBJID = c("1", "2"), AGE = c(34, 56))
anl_data <- tibble(USUBJID = "1", AGE = 34, AVAL = c(12, 17))
derive_summary_records(
  dataset_add = anl_data,
  dataset_ref = ref_data,
  by_vars = exprs(USUBJID),
  set_values_to = exprs(
    AVAL = mean(AVAL),
    DTYPE = "AVERAGE"
  )
)
```
results in 
```
# A tibble: 2 × 4
  USUBJID  AVAL DTYPE     AGE
  <chr>   <dbl> <chr>   <dbl>
1 1        14.5 AVERAGE    NA
2 2        NA   NA         56
```

### PR DESCRIPTION
**Pull request by manciniedoardo**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/derive_param_exist_flag.R
Comment (bundfussr) [inline R/derive_param_exist_flag.R:1]: Could you remove the default values for `true_value`, `false_value`, and `missing_value` from the argument descriptions?

File: R/derive_summary_records.R [hunk 1]
```diff
@@ -273,11 +273,16 @@ derive_summary_records <- function(dataset = NULL,
...
     ref_vars <- colnames(dataset_ref)
+    by_vars_chr <- map_chr(by_vars, as_name)
+
+    common_vars <- add_vars %>%
+      intersect(ref_vars) %>%
+      intersect(by_vars_chr)
 
     new_ref_obs <- anti_join(
-      select(dataset_ref, intersect(add_vars, ref_vars)),
+      select(dataset_ref, all_of(common_vars)),
```
[the remark this answers is not in the document]
Comment (manciniedoardo) [reply R/derive_summary_records.R]: You're right, I was playing around with various options just to make sure I didn;'t miss anything and it looks like I was just too conservative.

File: R/derive_param_exist_flag.R
Comment (manciniedoardo) [reply R/derive_param_exist_flag.R:1]: Done

File: R/derive_summary_records.R [hunk 2]
```diff
@@ -275,7 +275,7 @@ derive_summary_records <- function(dataset = NULL,
     ref_vars <- colnames(dataset_ref)
```
Comment (bundfussr) [inline R/derive_summary_records.R]: Please remove `add_vars` and `ref_vars` as they are no longer used.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Review by bundfussr — APPROVED** [review]

**Comment by manciniedoardo** [comment]
@bms63 good to go or wanna take a look?

**Comment by bms63** [comment]
does a new test need to be written for this scenario or was it already covered?

**Comment by manciniedoardo** [comment]
> does a new test need to be written for this scenario or was it already covered?

i think it should now be covered by existing suite of tests
